crates/sui-core/src/authority/authority_store.rs

```diff
@@ -910,8 +964,20 @@ impl<
         let index_to_write = std::iter::once((LAST_CONSENSUS_INDEX_ADDR, consensus_index));
 
         // Atomically store all elements.
+        //
+        // certificate_to_write is written as part of this batch because it can be, not because it
+        // needs to be stored atomically with sequenced/schedule/last_consensus_index.
+        //
+        // TODO: We store certificate_to_write so that we can resume processing of this shared
+        // tx later if necessary. Therefore, instead of writing it to self.certificates (which is
+        // traditionally where we store certificates that have been fully processed), we should
+        // write this to a table/store specifically for in-progress shared-object txes, from which it will be
```

Comment:
Do in-progress certificates even need to be persisted?   Usually, recovery of a node involves replaying of transactions which have not been finished, at the very least, and those are stored in the transactions table, plus the sequenced_* tables store which transactions have been scheduled to be executed.  Those seem enough to me. 

Part of thinking about storage design is figuring out what things really need to be persisted, and it is usually much less than initially thought, once we consider recovery semantics.

Reply:
This comment will probably be subsumed by https://www.notion.so/mystenlabs/Safe-TX-storage-and-sequencing-83e94806fac34910b867b5b9352fe698 - In any case I was mainly bringing over comments from here: https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority.rs#L1153